.github/workflows/windows-periodic.yml

```diff
@@ -0,0 +1,156 @@
+                            git checkout $SHIM_COMMIT ; go build -mod=vendor -o ${{ env.REMOTE_VM_BIN_PATH }}\containerd-shim-runhcs-v1.exe .\cmd\containerd-shim-runhcs-v1"
+
+      - name: RunIntegrationTests
+        run: |
+          ssh -i $HOME/.ssh/id_rsa ${{ env.SSH_OPTS }} azureuser@${{ env.VM_PUB_IP }} "sh.exe -c 'cd /c/containerd && (make integration | tee /c/Logs/integration.log)'"
+          ssh -i $HOME/.ssh/id_rsa ${{ env.SSH_OPTS }} azureuser@${{ env.VM_PUB_IP }} "sh.exe -c 'cat /c/Logs/integration.log | go-junit-report.exe > /c/Logs/junit_00.xml'"
+
+      - name: PrepareRepoList
+        run: |
+          ssh -i $HOME/.ssh/id_rsa ${{ env.SSH_OPTS }} azureuser@${{ env.VM_PUB_IP }} "sh -c 'cat > c:/repolist.yaml'" <<'EOF'
+            dockerLibraryRegistry: docker.io/claudiubelu
+            k8sCriContainerdRegistry: docker.io/claudiubelu
+            k8sGcrRegistry: docker.io/claudiubelu
+          EOF          
+
+      - name: RunCRIIntegrationTests            
+        run: |
+          ssh -i $HOME/.ssh/id_rsa ${{ env.SSH_OPTS }} azureuser@${{ env.VM_PUB_IP }} "sh.exe -s" <<EOF
+            cd c:/containerd
+            ./script/setup/install-cni-windows
+            export KUBE_TEST_REPO_LIST=c:/repolist.yaml
+            make cri-integration | tee c:/Logs/cri-integration.log
+          EOF
+          ssh -i $HOME/.ssh/id_rsa ${{ env.SSH_OPTS }} azureuser@${{ env.VM_PUB_IP }} "sh.exe -c 'cat c:\Logs\cri-integration.log | go-junit-report.exe > c:/Logs/junit_01.xml' "
+
+      - name: PullLogsFromWinNode
+        run: |
+          scp -i $HOME/.ssh/id_rsa ${{ env.SSH_OPTS }} azureuser@${{ env.VM_PUB_IP }}:c:/Logs/*.xml ${{ env.LOGS_DIR }}/artifacts/
+          for f in $(ls ${{ env.LOGS_DIR }}/artifacts/*.xml); do
+              xmlstarlet ed -d "/testsuites/testsuite/properties" $f > ${{ env.LOGS_DIR }}/$(basename $f)
+              mv ${{ env.LOGS_DIR }}/$(basename $f) $f
+          done
+
+      - name: FinishJob
+        run: |
+          jq -n --arg result SUCCESS --arg timestamp $(date +%s) '$timestamp|tonumber|{timestamp:.,$result}' > ${{ env.LOGS_DIR }}/finished.json
+          echo "${{ env.STARTED_TIME }}" > $HOME/latest-build.txt
+
+      - name: SetUpCloudSDK
+        uses: google-github-actions/setup-gcloud@master
+        with:
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
+          service_account_key: ${{ secrets.GCP_SA_KEY }}
+          export_default_credentials: true
+
+      - name: UploadArtifacts
+        run: |
+          gsutil cp -r ${{ env.LOGS_DIR }} ${{ env.GOOGLE_BUCKET }}      
+          gsutil cp $HOME/latest-build.txt ${{ env.GOOGLE_BUCKET }}
+
+      - name: ResourceCleanup
+        uses: azure/CLI@v1
+        with:
+        azcliversion: latest
+        inlinescript: |
+          az group delete -g ${{ env.AZURE_RESOURCE_GROUP }} --yes
```

Comment:
Could we add https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#always flag so the RG always gets cleaned up in the event other steps fail?

Reply:
Will do